x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/IndexFollowingIT.java

```diff
@@ -981,9 +988,63 @@ public void testMustCloseIndexAndPauseToRestartWithPutFollowing() throws Excepti
     }
 
     public void testIndexFallBehind() throws Exception {
+        runFallBehindTest(
+                () -> {
+                    // we have to remove the retention leases on the leader shards to ensure the follower falls behind
+                    final ClusterStateResponse followerIndexClusterState =
+                            followerClient().admin().cluster().prepareState().clear().setMetaData(true).setIndices("index2").get();
+                    final String followerUUID = followerIndexClusterState.getState().metaData().index("index2").getIndexUUID();
+                    final ClusterStateResponse leaderIndexClusterState =
+                            leaderClient().admin().cluster().prepareState().clear().setMetaData(true).setIndices("index1").get();
+                    final String leaderUUID = leaderIndexClusterState.getState().metaData().index("index1").getIndexUUID();
+
+                    final RoutingTable leaderRoutingTable = leaderClient()
+                            .admin()
+                            .cluster()
+                            .prepareState()
+                            .clear()
+                            .setIndices("index1")
+                            .setRoutingTable(true)
+                            .get()
+                            .getState()
+                            .routingTable();
+
+                    for (ObjectCursor<IndexShardRoutingTable> shardRoutingTable : leaderRoutingTable.index("index1").shards().values()) {
+                        final ShardId shardId = shardRoutingTable.value.shardId();
+                        leaderClient().execute(
+                                RetentionLeaseActions.Remove.INSTANCE,
+                                new RetentionLeaseActions.RemoveRequest(shardId, retentionLeaseId(followerUUID, leaderUUID)))
+                                .get();
+                    }
+                },
+                exceptions -> assertThat(exceptions.size(), greaterThan(0)));
+    }
+
+    public void testIndexDoesNotFallBehind() throws Exception {
+        runFallBehindTest(
+                () -> {},
+                exceptions -> assertThat(exceptions.size(), equalTo(0)));
+    }
+
+    /**
+     * Runs a fall behind test. In this test, we construct a situation where a follower is paused. While the follower is paused we index
+     * more documents that was causes soft deletes into the leader, flush them, and run a force merge. This is to set up a situation where
```

Comment:
that was

Reply:
I pushed 0ac55338bc5dbdb32300890a341d5e6a98e91f1b.